### PR TITLE
[portBLAS] Add USM support for rotmg

### PR DESCRIPTION
**Pull request by aacostadiaz**
This pull request adds USM support for the rotmg operator and updates the test to align it with the Buffer test.

- [X] Do all unit tests pass locally? [TestResults_level1_rotmg.html.zip](https://github.com/oneapi-src/oneMKL/files/12474668/TestResults_level1_rotmg.html.zip)
- [X] Have you formatted the code using clang-format?

### CODE REVIEW

**Review by muhammad-tanvir-1211 — APPROVED** [review]
LGTM, Thanks.

File: src/blas/backends/portblas/portblas_level1.cxx [hunk 1]
```diff
@@ -331,7 +331,16 @@ sycl::event rotm(sycl::queue &queue, std::int64_t n, real_t *x, std::int64_t inc
...
                   const std::vector<sycl::event> &dependencies) {
-    throw unimplemented("blas", "rotmg", " for USM");
+    auto y_d =
+        (real_t *)sycl::malloc_device(sizeof(real_t), queue.get_device(), queue.get_context());
+    auto copy_in_event = queue.memcpy(y_d, &y1, sizeof(real_t), dependencies);
+    auto rotmg_event = std::invoke([&]() -> sycl::event {
+        CALL_PORTBLAS_USM_FN(::blas::_rotmg, queue, d1, d2, x1, y_d, param,
+                             std::vector<sycl::event>{ copy_in_event });
+    });
+    rotmg_event.wait();
```
Comment (mkrainiuk) [inline src/blas/backends/portblas/portblas_level1.cxx]: Should we document somewhere that async execution is not supported?

**Review by mkrainiuk — APPROVED** [review]
Looks good to me, thanks.

**Comment by muhammad-tanvir-1211** [comment]
Hi @andrewtbarker, could you please take a look at this PR and share feedback? Thanks.

File: src/blas/backends/portblas/portblas_level1.cxx [hunk 2]
```diff
@@ -331,7 +331,18 @@ sycl::event rotm(sycl::queue &queue, std::int64_t n, real_t *x, std::int64_t inc
...
                   const std::vector<sycl::event> &dependencies) {
-    throw unimplemented("blas", "rotmg", " for USM");
+    auto y_d =
+        (real_t *)sycl::malloc_device(sizeof(real_t), queue.get_device(), queue.get_context());
+    auto copy_in_event = queue.memcpy(y_d, &y1, sizeof(real_t), dependencies);
+    auto rotmg_event = std::invoke([&]() -> sycl::event {
+        CALL_PORTBLAS_USM_FN(::blas::_rotmg, queue, d1, d2, x1, y_d, param,
+                             std::vector<sycl::event>{ copy_in_event });
+    });
+    auto free_event = queue.submit([&](cl::sycl::handler &cgh) {
```
Comment (andrewtbarker) [inline src/blas/backends/portblas/portblas_level1.cxx]:
```suggestion
    auto free_event = queue.submit([&](sycl::handler &cgh) {
```

**Review by andrewtbarker — APPROVED** [review]
Approved with very minor suggestion for change.

**Pull request merged**